pages/api/student_email_join.js

```diff
@@ -1,37 +1,51 @@
 import prisma from '../../prisma/prisma';
+import { unstable_getServerSession } from 'next-auth';
+import { authOptions } from './auth/[...nextauth]';
 
 export default async function handle(req, res) {
+  //unstable_getServerSession is recommended here: https://next-auth.js.org/configuration/nextjs
+  const session = await unstable_getServerSession(req, res, authOptions);
+
+  if (!req.method == 'POST') {
+    res.status(405).end();
+  }
+
+  if (!session) {
+    res.status(403).end();
+  }
+
+  let user = await prisma.user.findUniqueOrThrow({
```

Comment:
Is this what you meant by pulling from databse instead of querying API? @roberthuskins